plugins/net_plugin/net_plugin.cpp

```diff
@@ -589,6 +589,8 @@ namespace eosio {
       optional<request_message>   last_req;
       handshake_message           last_handshake_recv;
       handshake_message           last_handshake_sent;
+      string                      last_status;
+      uint32_t                    last_block{0};
```

Comment:
This is used both for last block sent and last block received. I think we should track them separately. Also probably better to make them both atomic and move out of this mutex (not comment above says these are covered by mutex `conn_mtx`.

Reply:
Also the enum for last_status could be an atomic as well and just avoid all the use `conn_mtx`.